### ISSUE TITLE
Error: Failed to list ipv4 rules for LXD network test (table mangle)

### ISSUE THREAD

**Issue by quebin31** [lxc/lxd#5840]
# Required information

 * Distribution: Arch Linux
 * Distribution version: Not aplicable
 * The output of "lxc info" or if that fails:
```
config: {}
api_extensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_stop_priority
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
- storage
- file_delete
- file_append
- network_dhcp_expiry
- storage_lvm_vg_rename
- storage_lvm_thinpool_rename
- network_vlan
- image_create_aliases
- container_stateless_copy
- container_only_migration
- storage_zfs_clone_copy
- unix_device_rename
- storage_lvm_use_thinpool
- storage_rsync_bwlimit
- network_vxlan_interface
- storage_btrfs_mount_options
- entity_description
- image_force_refresh
- storage_lvm_lv_resizing
- id_map_base
- file_symlinks
- container_push_target
- network_vlan_physical
- storage_images_delete
- container_edit_metadata
- container_snapshot_stateful_migration
- storage_driver_ceph
- storage_ceph_user_name
- resource_limits
- storage_volatile_initial_source
- storage_ceph_force_osd_reuse
- storage_block_filesystem_btrfs
- resources
- kernel_limits
- storage_api_volume_rename
- macaroon_authentication
- network_sriov
- console
- restrict_devlxd
- migration_pre_copy
- infiniband
- maas_network
- devlxd_events
- proxy
- network_dhcp_gateway
- file_get_symlink
- network_leases
- unix_device_hotplug
- storage_api_local_volume_handling
- operation_description
- clustering
- event_lifecycle
- storage_api_remote_volume_handling
- nvidia_runtime
- container_mount_propagation
- container_backup
- devlxd_images
- container_local_cross_pool_handling
- proxy_unix
- proxy_udp
- clustering_join
- proxy_tcp_udp_multi_port_handling
- network_state
- proxy_unix_dac_properties
- container_protection_delete
- unix_priv_drop
- pprof_http
- proxy_haproxy_protocol
- network_hwaddr
- proxy_nat
- network_nat_order
- container_full
- candid_authentication
- backup_compression
- candid_config
- nvidia_runtime_config
- storage_api_volume_snapshots
- storage_unmapped
- projects
- candid_config_key
- network_vxlan_ttl
- container_incremental_copy
- usb_optional_vendorid
- snapshot_scheduling
- container_copy_project
- clustering_server_address
- clustering_image_replication
- container_protection_shift
- snapshot_expiry
- container_backup_override_pool
- snapshot_expiry_creation
- network_leases_location
- resources_cpu_socket
- resources_gpu
- resources_numa
- kernel_features
- id_map_current
- event_location
- storage_api_remote_volume_snapshots
- network_nat_address
- container_nic_routes
- rbac
- cluster_internal_copy
- seccomp_notify
- lxc_features
- container_nic_ipvlan
api_status: stable
api_version: "1.0"
auth: trusted
public: false
auth_methods:
- tls
environment:
  addresses: []
  architectures:
  - x86_64
  - i686
  certificate: |
    -----BEGIN CERTIFICATE-----
    MIIB+DCCAX2gAwIBAgIQE9UPeeNd7U/kdveeSt0lMjAKBggqhkjOPQQDAzAyMRww
    GgYDVQQKExNsaW51eGNvbnRhaW5lcnMub3JnMRIwEAYDVQQDDAlyb290QGFyY2gw
    HhcNMTkwNjEyMTU0MjUxWhcNMjkwNjA5MTU0MjUxWjAyMRwwGgYDVQQKExNsaW51
    eGNvbnRhaW5lcnMub3JnMRIwEAYDVQQDDAlyb290QGFyY2gwdjAQBgcqhkjOPQIB
    BgUrgQQAIgNiAAQRcPD+5N8lsQOnD5pMSn4M6HdiGhwYyYqfTRvb03GC0qyZyCwa
    tYrqynYK2xuMkkwJKme5fXzyDd8lnHR9x5++pRnx0S3vkYUD8MiMqUZsHfWwhO6C
    UrYUbyl6hbaboNWjWDBWMA4GA1UdDwEB/wQEAwIFoDATBgNVHSUEDDAKBggrBgEF
    BQcDATAMBgNVHRMBAf8EAjAAMCEGA1UdEQQaMBiCBGFyY2iHBMCoAQyHBKwSAAGH
    BKwRAAEwCgYIKoZIzj0EAwMDaQAwZgIxAMQnqSUFCfs5EgYZ208nYU1Asm3dzJOy
    Wkj+Sr4UN1PiHwMhVmKk7xZO7pMs7cKh4AIxAMv4yR51mkDcMKb9gMRgMd8F7IHm
    f2XDBYT1hs3yhgs5MlGIeZoFqf7AIch+mPtEXQ==
    -----END CERTIFICATE-----
  certificate_fingerprint: 8614b468704d802a19d7c049dfa05a8b75476910ea21479b14c86eba12d11bab
  driver: lxc
  driver_version: 3.1.0
  kernel: Linux
  kernel_architecture: x86_64
  kernel_features:
    netnsid_getifaddrs: "true"
    seccomp_listener: "true"
    shiftfs: "false"
    uevent_injection: "true"
    unpriv_fscaps: "true"
  kernel_version: 5.1.7-arch1-1-ARCH
  lxc_features:
    mount_injection_file: "false"
    network_gateway_device_route: "false"
    network_ipvlan: "false"
    network_l2proxy: "false"
    seccomp_notify: "false"
  project: default
  server: lxd
  server_clustered: false
  server_name: arch
  server_pid: 30327
  server_version: "3.13"
  storage: ""
  storage_version: ""
```

# Issue description

When I executed `sudo lxd init`, the setup failed with the error without custom values in the configuration setup.

`Error: Failed to list ipv4 rules for LXD network test (table mangle)`

# Steps to reproduce

 1. `systemctl start lxd.service`
 2. `sudo lxd init`
 3. Hit Enter until program hangs configuring, wait for the error

# Information to attach

 - [ ] Any relevant kernel output (`dmesg`)
 - [ ] Container log (`lxc info NAME --show-log`)
 - [ ] Container configuration (`lxc config show NAME --expanded`)
 - [ ] Main daemon log (at /var/log/lxd/lxd.log or /var/snap/lxd/common/lxd/logs/lxd.log)
 - [ ] Output of the client with --debug
 - [ ] Output of the daemon with --debug (alternatively output of `lxc monitor` while reproducing the issue)

**Comment by stgraber** [comment]
What does `iptables -t mangle -L -n` show on your system?

**Comment by stgraber** [comment]
And how was LXD installed? is that the native package or the snap?

**Comment by antifob** [comment]
I had this bug before. The solution is to load the `ip_tables` and `ip6_tables` kernel modules.

**Comment by stgraber** [comment]
@quebin31 does that work for you? ^

**Comment by quebin31** [comment]
I'm not able to reproduce the error again.
I installed the package as pre-built binary from archlinuxcn repository at first, after the error y removed the package and everything in `/var/lib/lxd`.
Then I installed `lxd` from the AUR, and then I did `lxd init` and everything worked normally, I think it may be a problem with the binary pre-built from archlinuxcn repo. If you still need info about my iptables just tell me.
